test/client/components/script/script.test.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import renderer from 'react-test-renderer';
+import ScriptPage from '../../../../public/src/components/script/script';
+import { fetchScript, fetchScriptQuestions } from '../../../../public/src/actions/script';
+
+const props = {
+  script_id: 1,
+  script_info: {
+    name: 'ScriptName',
+    body: 'ScriptBody',
+    description: 'ScriptDescription'
+  },
+  questions: [
+    {
+      id: 4,
+      title: 'Age',
+      description: 'What is your age range?',
+      type: 'singleselect',
+      responses: '0-10,11-20,21-40,41-60,61+',
+      created_at: '2017-08-11T21:36:45.387Z',
+      updated_at: '2017-08-11T21:36:45.387Z',
+      script_id: 1,
+      question_id: 3,
+      sequence_number: 1
+    },
+    {
+      id: 5,
+      title: 'Hobbies',
+      description: 'What are your hobbies?',
+      type: 'multiselect',
+      responses: 'swimming,running,biking,sleeping,eating,weaving',
+      created_at: '2017-08-11T21:36:45.398Z',
+      updated_at: '2017-08-11T21:36:45.398Z',
+      script_id: 1,
+      question_id: 2,
+      sequence_number: 2
+    },
+    {
+      id: 6,
+      title: 'Ice Cream',
+      description: 'How do you feel about ice cream?',
+      type: 'paragraph',
+      responses: null,
+      created_at: '2017-08-11T21:36:45.514Z',
+      updated_at: '2017-08-11T21:36:45.514Z',
+      script_id: 1,
+      question_id: 1,
+      sequence_number: 3
+    }
+  ],
+  fetchScript,
+  fetchScriptQuestions
+};
+
+describe('<ScriptPage />', () => {
+  it('renders correctly', () => {
+    const tree = renderer.create(
+        <ScriptPage {...props} />
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
+describe('ScriptPage props and methods', () => {
+  const wrapper = shallow(<ScriptPage {...props} />).instance();
+  describe('renderQuestions method', () => {
+    it('should have a method named renderQuestions', () => {
+      const { renderQuestions } = wrapper;
+      expect(typeof renderQuestions).toBe('function');
+    });
+    it('should return questions properly', () => {
+      const questions = wrapper.renderQuestions();
+      expect(Array.isArray(questions)).toBe(true);
+    });
+  });
+  describe('props', () => {
+    describe('script_id prop', () => {
+      it('should be defined', () => {
+        expect(props.script_id).toBeDefined();
+      });
+      it('should have a value of 1', () => {
+        expect(props.script_id).toBe(1);
+      });
+    });
+    describe('script_info prop', () => {
+      it('should be defined', () => {
+        expect(props.script_info).toBeDefined();
+      });
+      it('should be an object', () => {
+        expect(typeof props.script_info).toBe('object');
+      });
+      it('should have 3 properties - name, body, description', () => {
+        const scriptInfoKeys = Object.keys(props.script_info);
```

Comment:
I wrote a helper function for this if you want to take a look. Uses reduce and checks if the object has specific properties.